WordPress/MediaEditor/MediaEditor.swift

```diff
@@ -0,0 +1,34 @@
+import UIKit
+import TOCropViewController
+
+public class MediaEditor: NSObject {
+
+    let cropViewControllerFactory: (UIImage) -> TOCropViewController
+
+    private(set) var cropViewController: TOCropViewController?
+    private var callback: ((UIImage?) -> ())?
+
+    public init(cropViewControllerFactory: @escaping (UIImage) -> TOCropViewController = TOCropViewController.init) {
+        self.cropViewControllerFactory = cropViewControllerFactory
+        super.init()
+    }
+
+    public func edit(_ image: UIImage, from viewController: UIViewController? = nil, callback: @escaping (UIImage?) -> ()) {
+        self.callback = callback
+        cropViewController = self.cropViewControllerFactory(image)
+        cropViewController?.delegate = self
+        viewController?.present(cropViewController!, animated: true)
```

Comment:
Can you guard instead of force unwrap?

Reply:
If you look at the line `18` I just instantiated the cropViewController and `self.cropViewControllerFactory(image)` doesn't return an `Optional`.

I can use `guard` or `if` but it's a little bit strange:

```swift
cropViewController = self.cropViewControllerFactory(image)
cropViewController?.delegate = self
if let cropViewController = self.cropViewController {
    viewController?.present(cropViewController, animated: true)
}
```

or:

```swift
cropViewController = self.cropViewControllerFactory(image)
cropViewController?.delegate = self

guard let cropViewController = self.cropViewController else {
    return
}
viewController?.present(cropViewController, animated: true)
```

or another option:

```swift
let cropViewController = self.cropViewControllerFactory(image)
cropViewController.delegate = self
viewController?.present(cropViewController, animated: true)
self.cropViewController = cropViewController
```

What do you think?

Reply:
LGTM